src/control/ar/DeadzoneChecker.ts

```diff
@@ -139,12 +144,12 @@ class DeadzoneChecker {
       const diff = new THREE.Vector2().subVectors(input, prevPos);
       if (diff.length() > deadzone) {
         if (Math.abs(diff.x) > Math.abs(diff.y)) {
-          if (TOUCH.GESTURE.ONE_FINGER_HORIZONTAL & testingGestures) {
-            this._detectedGesture = TOUCH.GESTURE.ONE_FINGER_HORIZONTAL;
+          if (GESTURE.ONE_FINGER_HORIZONTAL & testingGestures) {
```

Comment:
Do you think you can wrap the if statement with testingGesture in the method code?

Reply:
I couldn't understand what you're trying to say.

Do you mean wrapping it like this?
```diff
-if (GESTURE.ONE_FINGER_HORIZONTAL & testingGestures) {
+if (this._hasGesture(GESTURE.ONE_FINGER_HORIZONTAL)) {
```

Or, do you mean to remove common codes like this?
```diff
const diff = new THREE.Vector2().subVectors(input, prevPos);
-if (diff.length() > deadzone) {
-  if (Math.abs(diff.x) > Math.abs(diff.y)) {
-    if (GESTURE.ONE_FINGER_HORIZONTAL & testingGestures) {
-      this._detectedGesture = GESTURE.ONE_FINGER_HORIZONTAL;
-    }
-  } else {
-    if (GESTURE.ONE_FINGER_VERTICAL & testingGestures) {
-      this._detectedGesture = GESTURE.ONE_FINGER_VERTICAL;
-    }
-  }
-}
+this._detectedGesture = this._checkGesture(diff, GESTURE.ONE_FINGER_HORIZONTAL, GESTURE.ONE_FINGER_VERTICAL);
```

Reply:
Sorry I misread the operator.

Reply:
But this code looks good.
```js
this._detectedGesture = this._checkGesture(diff, GESTURE.ONE_FINGER_HORIZONTAL, GESTURE.ONE_FINGER_VERTICAL);
```